Add `url` plugin option to support overriding the desired download URL

staticfloat opened the pull request:
This is useful when you want to use this plugin with some kind of internal build of Julia.  While it overrides things like the selected `version` and `arch`, it still has to do a bit of a dance to parse the URL out appropriately, only supports `https`, and may have other limitations.  I'm open to reimplementing this in a better way.

staticfloat committed:
Add `url` plugin option to support overriding the desired download URL

This is useful when you want to use this plugin with some kind of
custom build of Julia.  There's a little bit of annoying parsing we have
to do to fill out e.g. `bucket_host` and `path`, but overall I think
it's bearable.

I'm open to alternative implementations if we want to clean this up
further.

maleadt reviewed (approved):
SGTM in principle. Doing all this in Bash is starting to get messy though 🙂

staticfloat committed:
Always run with `debug_plugin: true` on CI

Also override `PS4` so that we don't conflate bash's prompts for
buildkite's headers.

staticfloat commented:
Agreed; if only we could do it in Julia..... ;)
